File: Sorting/Bubble_Sort.c
#include <stdio.h>

void swap(int*,int*);

int main()
{
 int num,i,j;
 printf("Enter Number Of Elements In Array : \n");
 scanf("%d",&num);
 int arr[num];
 printf("Enter Elements Of Array one after another : \n");
 for(i=0;i<num;i++)
  scanf("%d",&arr[i]);
 printf("Unsorted Array :- \n\n");
 for(i=0;i<num;i++)
  printf("%d\n",arr[i]);

 for(i=0;i<(num-1);i++)		//Bubble Sort Complexity O(n^2)
  for(j=0;j<(num-1-i);j++)
   {
    if(arr[j]>arr[j+1])
       swap(&arr[j+1],&arr[j]);
   }

 printf("Sorted Array :- \n\n");
 for(i=0;i<num;i++)
  printf("%d\n",arr[i]);

 return 0;
}

void swap(int *a, int*b)
{
 int temp=*a;
 *a=*b;
 *b=temp;
}

File: Searching and Sorting/Bubble_Sort.py
# O(n^2)

def Bubble(arr):

    swaps = 0

    for i in range(len(arr)):
        for j in range(len(arr) - 1):
            
            if(arr[j] >= arr[j+1]):
                swaps += 1
                arr[j], arr[j+1] = arr[j+1], arr[j]

        if(swaps == 0):
            break

    return arr


def main():
    ip = list(map(int, input("Enter the unsorted array: ").split()))

    sorted_arr = Bubble(ip)

    print("Sorted Array: ",end = "")
    print(*sorted_arr)

if __name__ == "__main__":
    main()